spec/helpers/http_mock/asserts.lua

```diff
@@ -0,0 +1,89 @@
+local helpers = require "spec.helpers"
+
+local setmetatable = setmetatable
+local ipairs = ipairs
+local unpack = unpack
+local pcall = pcall
+
+local http_mock = {}
+
+local step = {}
+
+local eventually_mt = {}
+
+function eventually_mt.__index(t, k)
+  local f = function(eventually_tbl, ...)
+    local mock = eventually_tbl.__mock
+    local args = {...}
+    -- todo: get correct level for stack trace
+    helpers.wait_until(function()
+      return step[k](mock, unpack(args))
+    end, mock.eventually_timeout)
+  end
+
+  t[k] = f
+  return f
+end
+
+function step.has_session(mock, f)
```

Comment:
Please write comments to explain what is the `session`.

Reply:
added to `register_assertion`.